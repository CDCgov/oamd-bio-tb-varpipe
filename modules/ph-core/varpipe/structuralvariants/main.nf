process VARPIPE_STRUCTURALVARIANTS {

    tag "${meta.id}"
    label 'process_single'

    container 'oamd-bio-biocontainers-python:3.12_80c558c_v3'

    input:
    tuple val(meta), path(qc_status), path(coverage), path(unmapped), path(mapped)
    tuple val(meta2), path(genome_region_coverage)
    path(bedstruct)

    output:
    tuple val(meta), path("*_structural_variants.txt"), emit: structural_variants, optional: true
    path("versions.yml"), emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    template 'structvar_parser.py'

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_stats.txt
    touch ${prefix}_target_region_coverage.txt
    touch ${prefix}_genome_region_coverage.txt
    touch ${prefix}_structural_variants.txt
    touch .qc_passed.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | cut -d' ' -f2)
    END_VERSIONS
    """
}
