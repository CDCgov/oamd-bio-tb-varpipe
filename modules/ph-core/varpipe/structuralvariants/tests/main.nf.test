nextflow_process {
    name "Test Process VARPIPE_STRUCTURALVARIANTS"
    script "../main.nf"
    process "VARPIPE_STRUCTURALVARIANTS"

    tag "modules"
    tag "modules_phcore"
    tag "varpipe"
    tag "varpipe/structuralvariants"

    test("varpipe structural variant analysis [coverages]") {
        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file('qc_passed.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_coverage.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_unmapped.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_mapped.txt')
                ]

                input[1] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_genome_region_coverage.txt')
                ]
                input[2] = file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/BED.txt')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("varpipe coverage analysis [bam] [coverages] -stub") {
        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
                    file('qc_passed.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_coverage.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_unmapped.txt'),
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/sample1_mapped.txt')
                ]

                input[1] = [
                    [ id:'test' ], // meta map
                    file(params.modules_testdata_base_path + '/genomics/tuberculosis/coverage/test_genome_region_coverage.txt')
                ]
                input[2] = file(params.modules_testdata_base_path + '/genomics/tuberculosis/genome/kb/varpipe/BED.txt')
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
