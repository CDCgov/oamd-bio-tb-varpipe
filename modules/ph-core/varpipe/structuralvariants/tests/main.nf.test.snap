{
    "varpipe coverage analysis [bam] [coverages] -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_structural_variants.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b39ed10974c19b79b2fa11c35c89dc60"
                ],
                "structural_variants": [
                    [
                        {
                            "id": "test"
                        },
                        "test_structural_variants.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b39ed10974c19b79b2fa11c35c89dc60"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-05-07T08:30:27.765410733"
    },
    "varpipe structural variant analysis [coverages]": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_structural_variants.txt:md5,adce0d578c66c214a53c5d2eabc4c625"
                    ]
                ],
                "1": [
                    "versions.yml:md5,274e4f906af29edb6755a1422f717204"
                ],
                "structural_variants": [
                    [
                        {
                            "id": "test"
                        },
                        "test_structural_variants.txt:md5,adce0d578c66c214a53c5d2eabc4c625"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,274e4f906af29edb6755a1422f717204"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.6"
        },
        "timestamp": "2025-05-07T15:43:54.756950466"
    }
}
